schedview/collect/consdb.py

```diff
@@ -11,7 +12,12 @@ def read_consdb(
     **kwargs,
 ) -> pd.DataFrame:
     consdb_visits = load_consdb_visits(instrument, *args, **kwargs)
-    visit_records: np.recarray = consdb_visits.opsim.to_records()
+    # If the return is empty, to_records won't be able to determine the
+    # right dtypes, so use ObservationArray()[0:0] instead to make the
+    # empty recarray with the correct dtypes.
+    visit_records: np.recarray = (
+        consdb_visits.opsim.to_records() if len(consdb_visits.opsim) > 0 else ObservationArray()[0:0]
+    )
```

Comment:
If it's empty, does anything else need to run?

Reply:
If I don't know exactly what the result will be used for, I find it often helps save trouble later on if I return an empty tables (whether represend as arecarray or DataFrame) with the expected columns and column types, so that it can be indexed or sliced. For example, if someone wants to count the number of visits with cos(fieldRA)>0.5 (I don't know why they might want to do this, it's just an example), they might do:

`print(len(visits.loc[np.cos(visits.fieldRA>0.5), :])`

If it is an empty DataFrame with a fieldRA of a type you can take a cosine of, this will work fine whether or not visits has any rows. But, if its None or a DataFrame with no fields defined, or if, say fieldRA is defined but a string, it will fail. In this case, though, there's no reason why it should fail. The same sort of thing happens with recarrays.

Reply:
Sounds good.